### PR TITLE
Allow multiple args in udevadm info

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/udev/udevadm-info.c [hunk 1]
```diff
@@ -22,6 +22,25 @@
 #include "udevadm.h"
 #include "udevadm-util.h"
 
+typedef enum action_type {
+        ACTION_QUERY,
+        ACTION_ATTRIBUTE_WALK,
+        ACTION_DEVICE_ID_FILE,
+} ActionType;
```
Comment (poettering) [inline src/udev/udevadm-info.c:29]: hmm, both a lowercase name and an uppercase name? maybe clean this up and just do `typedef enum ActionType {…} ActionType;`  how we do it usually?

File: src/udev/udevadm-info.c [hunk 2]
```diff
@@ -22,6 +22,25 @@
...
+        QUERY_NAME,
+        QUERY_PATH,
+        QUERY_SYMLINK,
+        QUERY_PROPERTY,
+        QUERY_ALL,
+} QueryType;
+
+bool arg_root = false;
+bool arg_export = false;
+const char *arg_export_prefix = NULL;
```
Comment (poettering) [inline src/udev/udevadm-info.c]: should all be static

File: src/udev/udevadm-info.c [hunk 3]
```diff
@@ -22,6 +22,25 @@
...
+
+
```
Comment (poettering) [inline src/udev/udevadm-info.c:42]: spurious newline

File: src/udev/udevadm-info.c [hunk 4]
```diff
@@ -223,8 +243,75 @@ static void cleanup_db(void) {
...
+                const char *node;
 
+                r = sd_device_get_devname(device, &node);
+                if (r < 0)
+                        return log_error_errno(r, "No device node found: %m");
+
+                if (arg_root)
+                        printf("%s\n", node);
+                else
+                        printf("%s\n", node + STRLEN("/dev/"));
```
Comment (poettering) [inline src/udev/udevadm-info.c]:
```c
printf ("%s\n", path_startswith(node, "/dev/"));
```

is a lot prettier and more robust I think

File: src/udev/udevadm-info.c [hunk 5]
```diff
@@ -223,8 +243,75 @@ static void cleanup_db(void) {
...
+                const char *devlink;
+                bool first = true;
+
+                FOREACH_DEVICE_DEVLINK(device, devlink) {
+                        if (!first)
+                                printf(" ");
+                        if (arg_root)
+                                printf("%s", devlink);
+                        else
+                                printf("%s", devlink + STRLEN("/dev/"));
```
Comment (poettering) [inline src/udev/udevadm-info.c]: as above

File: src/udev/udevadm-info.c [hunk 6]
```diff
@@ -223,8 +243,75 @@ static void cleanup_db(void) {
...
+
+                        first = false;
+                }
+                printf("\n");
```
Comment (poettering) [inline src/udev/udevadm-info.c]: not that i matters, but i'd have used fputc() here...

**Comment by poettering** [comment]
looks good to me, but a few comments above

(also, i the device unit hookup with udevadm trigger made me wonder if we shouldn't hook things the other way round too: when "systemctl reload" is invoked on a device unit it might make sense to issue a "change" uevent, too)

File: src/udev/udevadm-info.c (hunk 5 above)
Comment (keszybz) [reply src/udev/udevadm-info.c]: That's preexisting code...

File: src/udev/udevadm-info.c (hunk 6 above)
Comment (keszybz) [reply src/udev/udevadm-info.c]: Here too...

**Pull request merged**